packages/vuetify/src/composables/dimensions.ts

```diff
@@ -1,63 +1,42 @@
+// Utilities
 import { computed } from 'vue'
+import { convertToUnit } from '@/util/helpers'
+import propsFactory from '@/util/propsFactory'
 
 // Types
-import type { Prop } from 'vue'
+export interface DimensionProps {
+  height?: number | string
+  maxHeight?: number | string
+  maxWidth?: number | string
+  minHeight?: number | string
+  minWidth?: number | string
+  width?: number | string
+}
 
-// Utils
-import { convertToUnit, keys } from '@/util/helpers'
-import propsFactory from '@/util/propsFactory'
+export type DimensionKey = keyof DimensionProps
+export type DimensionStyles = Partial<Record<DimensionKey, string>>
 
-// Props
-const allDimensionsProps = {
-  height: {
-    type: [Number, String],
-  },
-  maxHeight: {
-    type: [Number, String],
-  },
-  maxWidth: {
-    type: [Number, String],
-  },
-  minHeight: {
-    type: [Number, String],
-  },
-  minWidth: {
-    type: [Number, String],
-  },
-  width: {
-    type: [Number, String],
-  },
-}
+export const makeDimensionProps = propsFactory({
+  height: { type: [Number, String] },
+  maxHeight: { type: [Number, String] },
+  maxWidth: { type: [Number, String] },
+  minHeight: { type: [Number, String] },
+  minWidth: { type: [Number, String] },
+  width: { type: [Number, String] },
+})
 
-// Types
-type PropValue = string | number | null | undefined
-type PropNames = keyof typeof allDimensionsProps
-
-// Effect
-export function dimensionsFactory<S extends PropNames> (...possibleProps: S[]) {
-  const selectedProps = possibleProps.length ? possibleProps : keys(allDimensionsProps) as S[]
-
-  const makeDimensionsProps = propsFactory(selectedProps.reduce((obj, prop) => {
-    obj[prop] = allDimensionsProps[prop]
-    return obj
-  }, {} as Record<S, Prop<PropValue>>))
-
-  const useDimensions = (props: Partial<Record<S, PropValue>>) => {
-    const dimensions = computed(() => {
-      return selectedProps.reduce((obj, key) => {
-        const value: PropValue = props[key]
-        if (value) {
-          obj.style[key] = convertToUnit(value)
-        }
-        return obj
-      }, { style: {} } as { style: Record<S, string | undefined> })
-    })
-
-    return { dimensions }
-  }
-
-  return {
-    makeDimensionsProps,
-    useDimensions,
-  }
+export function useDimensionStyles (props: DimensionProps) {
+  const dimensionStyles = computed(() => {
+    const properties = Object.keys(props) as DimensionKey[]
```

Comment:
There's no filter on `props` now, it will convert *all* of the component's props to units.

Reply:
Oh and there's a typed `keys` function in util/helpers that doesn't need type casting.